src/twisted/mail/test/test_smtp.py

```diff
@@ -669,6 +671,47 @@ def check(ignored):
 
         return self.loopback(server, client).addCallback(check)
 
+    def test_ESMTPSenderFactory_TLSError(self):
+        """
+        Attempting to connect to an ESMTP server which presents an invalid certificate
+        should produce a meaningful error.
```

Comment:
Love this docstring, very close to https://jml.io/pages/test-docstrings.html .

(It would be even nicer if it gave a crisper definition of "meaningful" but that is really nitpicking)